UI/Content View/Helpers/ViewWidthKey.swift

```diff
@@ -0,0 +1,8 @@
+import SwiftUI
+
+struct ViewWidthKey: PreferenceKey {
+    static var defaultValue: CGFloat = 0
+    static func reduce(value: inout CGFloat, nextValue: () -> CGFloat) {
+        value = nextValue()
```

Comment:
You can simply use a variable instead of this, since it simply overrides what was before.

Reduce with a `PreferenceKey` is used for collecting the changes.

Reply:
A `ViewModel` or an `Interactor` would be needed in the View, since the View is a struct itself.